src/Uno.UI/UI/Xaml/UIElement.Pointers.cs

```diff
@@ -231,6 +231,22 @@ internal void ClearPointerState()
 			ClearDragOver();
 		}
 
+#if UNO_HAS_MANAGED_POINTERS
+		internal void AbortGestureRecognizer()
+		{
+			var element = this;
+			while (element is not null)
+			{
+				if (element.IsGestureRecognizerCreated)
+				{
+					element.GestureRecognizer.CompleteGesture();
+				}
+
+				element = element.GetParent() as UIElement;
```

Comment:
You should consider native element only also here (on macOS we have `UNO_HAS_MANAGED_POINTERS` but still using native elements)

Reply:
You should be able to use `this.GetVisualAncestry()` with a `foreach` loop instead of the `while` loop.